### ISSUE TITLE
Bug: GRPO quickstart `max_completion_length=256` default silently breaks training

### ISSUE THREAD

**Issue by dhruvnigam93** [huggingface/trl#5697]
### Reproduction

I was trying to get started with trl using the example in the `README.md`

```python
# copy-paste from README, single GPU
from datasets import load_dataset
from trl import GRPOConfig, GRPOTrainer
from trl.rewards import accuracy_reward

dataset = load_dataset("trl-lib/DeepMath-103K", split="train")

training_args = GRPOConfig(
    output_dir="output",
    # all defaults: per_device_train_batch_size=8, num_generations=8, max_completion_length=256
    logging_steps=1,
)

trainer = GRPOTrainer(
    model="Qwen/Qwen2.5-0.5B-Instruct",
    reward_funcs=accuracy_reward,
    train_dataset=dataset,
    args=training_args,
)
trainer.train()
```

**Result:** Almost all completion hits exactly 256 tokens. The reward is 0 at most steps. 

```
Step  loss  grad_norm  reward_mean  clipped_ratio  mean_length
1     0     0          0            1.0            256
2     0     0          0            1.0            256
...
10    0     0          0            1.0            256
```

**Default `max_completion_length=256` causes near-total truncation on the recommended dataset.** The quickstart pairs `Qwen/Qwen2.5-0.5B-Instruct` with `DeepMath-103K`, but the default 256-token limit truncates 97-99% of completions. The model needs ~600-700 tokens to produce a complete math solution, so it is physically cut off before it can output `\boxed{...}` and receive any reward. The model sees zero gradients for almost the entire run initially.

**Fix:** Set `max_completion_length=1024`. Truncation drops from 97-99% to 8%. Reward appears within the first few steps.

## Evidence

I ran 3 training runs (100 steps each, all defaults: `per_device_train_batch_size=8`, `num_generations=8`, A100 80GB), varying only `max_completion_length`:

| `max_completion_length` | Mean Truncation Rate | Mean Reward | Peak GPU Memory |
|---|---|---|---|
| **256 (default)** | **98.4%** | **0.005** | **14.7 GB** |
| **1024** | **14.2%** | **0.114** | **32.4 GB** |
| **2048** | **0.6%** | **0.146** | **73.6 GB** |

### Truncation rate and reward over 100 training steps

<img width="1505" height="768" alt="Image" src="https://github.com/user-attachments/assets/1dc3fd41-2d19-4d58-89f3-929b99c2b7eb" />

Left: at 256 tokens, `completions/clipped_ratio` is pinned at ~100% for the entire run — the model almost never finishes a single answer. At 1024, truncation drops to ~14%. Right: at 256 tokens, reward flatlines at 0. At 1024, reward signal appears immediately. The same model, same dataset, same reward function produces ~23x more reward simply by allowing enough tokens to finish answering. Note the memory cost: going from 256 to 1024 more than doubles peak GPU memory (14.7 GB to 32.4 GB), and 2048 requires 73.6 GB — nearly filling an A100 80GB.

### System Info

- Platform: Linux-6.1.0-45-cloud-amd64-x86_64-with-glibc2.36
- Python version: 3.11.2
- TRL version: 1.1.0
- PyTorch version: 2.11.0
- accelerator(s): NVIDIA A100-SXM4-80GB
- Transformers version: 5.5.4
- Accelerate version: 1.13.0
- Datasets version: 4.8.4
- HF Hub version: 1.10.2

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by dhruvnigam93** [comment]
Happy to open a PR for this. Plan to:
1. Set `max_completion_length=1024` in the quickstart examples
2. Regenerate the reward curves in grpo_trainer.md - they're currently based on `Qwen2-0.5B-Instruct`, but the quickstart recommends Qwen2.5
3. Add a GRPO tab to the OOM troubleshooting section with memory guidance

lmk if there are any concerns with this plan

**Comment by qgallouedec** [comment]
Hey @dhruvnigam93, sorry for the slow reply, catching up on the backlog. This is fixed now, the default `max_completion_length` was bumped to 512 in your own follow-up fix.
Closing this as solved by #6264. Thanks for tracking it down and landing the fix!
